File: DemoVulnerable.java
import java.io.*;
import java.sql.*;

public class DemoVulnerable {
    // Hardcoded sensitive data
    private static final String SECRET_KEY = "mySuperSecretKey123";

    public static void main(String[] args) throws Exception {
        // 1. Command injection
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        System.out.print("Enter a shell command: ");
        String userCommand = reader.readLine();
        // Vulnerable: user input is directly passed to the shell
        Runtime.getRuntime().exec(userCommand);

        // 2. Path traversal
        System.out.print("Enter a filename to read: ");
        String filename = reader.readLine();
        File file = new File(filename); // Vulnerable: no sanitization
        BufferedReader fileReader = new BufferedReader(new FileReader(file));
        System.out.println("File contents:");
        String line;
        while ((line = fileReader.readLine()) != null) {
            System.out.println(line);
        }
        fileReader.close();

        // 3. SQL injection
        System.out.print("Enter user id: ");
        String userId = reader.readLine();
        Connection conn = DriverManager.getConnection("jdbc:sqlite:test.db");
        Statement stmt = conn.createStatement();
        // Vulnerable: user input concatenated into SQL query
        ResultSet rs = stmt.executeQuery("SELECT * FROM users WHERE id = '" + userId + "'");
        while (rs.next()) {
            System.out.println("User: " + rs.getString("username"));
        }
        rs.close();
        stmt.close();
        conn.close();

        // 4. Print hardcoded secret key
        System.out.println("Secret Key: " + SECRET_KEY);
    }
}

File: demo_vulnerable.py
import os

def insecure_eval(user_input):
    # Vulnerable to code injection
    return eval(user_input)

def insecure_file_read(filename):
    # Vulnerable to path traversal
    with open(filename, 'r') as f:
        return f.read()

def hardcoded_secret():
    # Hardcoded secret key (bad practice)
    secret_key = "supersecret123"
    return secret_key

def sql_injection(cursor, user_id):
    # Vulnerable to SQL injection
    query = "SELECT * FROM users WHERE id = '%s'" % user_id
    cursor.execute(query)
    return cursor.fetchall()

def command_injection(user_input):
    # Vulnerable to command injection
    os.system("echo " + user_input)

# Example usage (for demonstration only - do not use in production!)
if __name__ == "__main__":
    print("Eval:", insecure_eval(input("Enter code to eval: ")))
    print("File read:", insecure_file_read(input("Enter filename to read: ")))
    print("Hardcoded secret:", hardcoded_secret())
    # The following require a database connection and proper setup to test:
    # print("SQL:", sql_injection(cursor, input("Enter user id: ")))
    print("Command injection test:")
    command_injection(input("Enter a command: "))
